pyk/src/pyk/kast.py

```diff
@@ -132,7 +132,7 @@ def update(self, atts: Mapping[str, Any]) -> 'KAtt':
 EMPTY_ATT: Final = KAtt()
 
 
-class WithKAtt(KAst, ABC):
+class WithKAtt(ABC):
```

Comment:
I don't yet know the purpose of the `WithKAtt` class, but this looks suspiciously like a mistake. Should `WithKAttr` inherit  from `KAst` instead?

Reply:
`WithKAtt` is a way to add an attribute `att: KAtt` to a class. The subclass implements `let_att` (i.e. defines how to "set" the `att`), and gets `map_att` and `update_atts` for free.

> Should `WithKAttr` inherit from `KAst` instead?

It could (with the intended meaning that whatever has an attribute `att` is also a `KAst`), but it is not necessary for `WithKAtt` to work.  Also, then implementing `WithKatt` would make it necessary to implement `KAst` as well.